data-science-onramp/ai-platform/tfkeras_test.py

```diff
@@ -0,0 +1,90 @@
+import os
+import tarfile
+import time
+import uuid
+
+from google.cloud import storage
+from google.cloud.aiplatform import gapic as aip
+from googleapiclient import discovery
+import pytest
+
+STAGING_BUCKET = f"tfkeras-job-dir-{uuid.uuid4()}"
+INPUT_DIR = "inputs"
+TRAINER_DIR = "modules"
+MODEL_DIR = "model"
+TRAIN_DATA = "train.csv"
+TRAINER_TAR = "trainer.tar.gz"
+TRAIN_DATA_PATH = f"gs://{STAGING_BUCKET}/{INPUT_DIR}/{TRAIN_DATA}"
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+MODEL_NAME = f"tfkeras-test-{uuid.uuid4()}"
+JOB_ID = f"tfkeras_{str(uuid.uuid4())[:7]}"
+DEPLOY_IMAGE = "gcr.io/cloud-aiplatform/training/tf-cpu.2-3:latest"
+
+TERMINAL_STATES = [
+    aip.JobState.JOB_STATE_SUCCEEDED,
+    aip.JobState.JOB_STATE_FAILED,
+    aip.JobState.JOB_STATE_CANCELLING,
+    aip.JobState.JOB_STATE_CANCELLED
+]
+
+
+@pytest.fixture(autouse=True)
+def setup_teardown():
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(STAGING_BUCKET, location=REGION)
+    bucket.blob(f"{INPUT_DIR}/{TRAIN_DATA}").upload_from_filename(TRAIN_DATA, timeout=600)
+
+    with tarfile.open(TRAINER_TAR, mode="x:gz") as tar:
+        tar.add(f"{TRAINER_DIR}/")
+
+    bucket.blob(TRAINER_TAR).upload_from_filename(TRAINER_TAR)
+
+    yield bucket
+
+    # [blob.delete() for blob in bucket.list_blobs()]
+    # bucket.delete()
```

Comment:
Thanks! Great stuff.

I think we're just missing code to cleanup the artifacts when done.
- Remove the training job
- Remove the trained model

I think the uCAIP notebook I sent you should have example code for cleaning those up.